pkg/webhook/util/filter.go

```diff
@@ -20,3 +24,21 @@ func HasInProgressingVMBackupBySourceUID(cache ctlharvesterv1.VirtualMachineBack
 	}
 	return false, nil
 }
+
+func HasInProgressingVMRestoreOnSameTarget(cache ctlharvesterv1.VirtualMachineRestoreCache, targetNamespace, targetName string) (bool, error) {
+	vmRestores, err := cache.GetByIndex(indexeres.VMRestoreByTargetNamespaceAndName, fmt.Sprintf("%s-%s", targetNamespace, targetName))
+	if err != nil && !apierrors.IsNotFound(err) {
+		return false, err
+	}
+
+	for _, vmRestore := range vmRestores {
+		if vmRestore != nil && vmRestore.Status != nil {
```

Comment:
Just curious to know what situation would make `vmRestore` nil if they are already inited?

Reply:
I don't think there is `nil` vmrestore. However, the return type for `cache.GetByIndex` is `[]*harvesterv1.VirtualMachineRestore`, so I would like to add the check to prevent panic in Golang.